src/org/freedesktop/gstreamer/Sample.java

```diff
@@ -40,24 +41,52 @@
     }
 
     /**
-     * Get the {@link Caps} associated with sample, or NULL when there is no
-     * caps. The caps remain valid as long as sample is valid.
+     * Get the {@link Caps} associated with sample, or NULL when there is no caps. 
+     * The caps remain valid as long as sample is valid.
+     * If you need to hold on to the caps for longer than that, take a ref to the caps 
      *
      * @return caps of sample or NULL when there is no caps.
      */
     public Caps getCaps() {
-        return GSTMESSAGE_API.gst_sample_get_caps(this);
+        return GSTSAMPLE_API.gst_sample_get_caps(this);
+    }
+    
+    /**
+     * Set the {@link Caps} associated with sample. 
+     * This sample must be writable.
+     * 
+     * Since GStreamer 1.16
+     * 
+     * @param caps
+     */
+    @Gst.Since(minor = 16)
+    public void setCaps(Caps caps) {
+    	GSTSAMPLE_API.gst_sample_set_caps(this, caps);
```

Comment:
Please add `Gst.checkVersion(1, 16);` as first line of this method. eg. like https://github.com/gstreamer-java/gst1-java-core/blob/master/src/org/freedesktop/gstreamer/Buffer.java#L250  Same for all methods requiring above 1.8.